aws/resource_aws_security_group_rule.go

```diff
@@ -842,20 +842,34 @@ func descriptionFromIPPerm(d *schema.ResourceData, rule *ec2.IpPermission) strin
 		}
 	}
 
-	// probe UserIdGroupPairs
-	groupIds := make(map[string]bool)
 	if raw, ok := d.GetOk("source_security_group_id"); ok {
-		groupIds[raw.(string)] = true
-	}
+		components := strings.Split(raw.(string), "/")
 
-	if len(groupIds) > 0 {
-		for _, gp := range rule.UserIdGroupPairs {
-			if _, ok := groupIds[*gp.GroupId]; !ok {
-				continue
+		switch len(components) {
+		case 2:
+			userId := components[0]
+			groupId := components[1]
+
+			for _, gp := range rule.UserIdGroupPairs {
+				if *gp.GroupId != groupId || *gp.UserId != userId {
+					continue
+				}
+				if *gp.GroupId == groupId && *gp.UserId == userId {
```

Comment:
i believe you can omit this logical check as only a `False || False` result in the above check will get us here

Reply:
ah, yes